### PR TITLE
Add success and error metrics to every worker

### PR DESCRIPTION
**Pull request by dreis0**
In order to have better tracking of all the worker's success and error rates, this PR adds `start`, `completed` and `error` metrics for each of the workers.

### CODE REVIEW

File: worker/process_batch_worker.go [hunk 1]
```diff
@@ -282,14 +282,25 @@ func (b *ProcessBatchWorker) Process(message *workers.Msg) {
...
+	b.checkErr(job, err)
 	log.D(l, "Updated job users info successfully.")
 	if float64(batchErrorCounter)/float64(len(parsed.Users)) > b.Workers.Config.GetFloat64("workers.processBatch.maxUserFailureInBatch") {
-		b.incrFailedBatches(job.ID, job.TotalBatches, parsed.AppName)
-		checkErr(l, fmt.Errorf("failed to send message to several users, considering batch as failed"))
+		b.incrFailedBatches(job, parsed.AppName)
+		b.checkErr(job, fmt.Errorf("failed to send message to several users, considering batch as failed"))
 	}
+
+	b.Workers.Statsd.Incr(processBatchWorkerCompleted, job.Labels(), 1)
```
Comment (bitbanshee) [inline worker/process_batch_worker.go:295]: Should we not cover all these returns?
* https://github.com/topfreegames/marathon/pull/51/files#diff-77f6b43c0608b2514ab8d6038ce06ec4e1961bd168589954a328405d9e1881cbR185
* https://github.com/topfreegames/marathon/pull/51/files#diff-77f6b43c0608b2514ab8d6038ce06ec4e1961bd168589954a328405d9e1881cbR189-R199

File: worker/metrics.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+	processBatchWorkerStart     = "starting_process_batch_worker"
+	processBatchWorkerCompleted = "completed_process_batch_worker"
+	processBatchWorkerError     = "error_process_batch_worker"
+
+	resumeJobWorkerStart     = "starting_resume_job_worker"
+	resumeJobWorkerCompleted = "completed_resume_job_worker"
+	resumeJobWorkerError     = "error_resume_job_worker"
+
+	getCsvFromS3Timing = "get_csv_from_s3"
```
Comment (bitbanshee) [inline worker/metrics.go:28]: All constants must start with upper case letter.

File: worker/resume_job_worker.go [hunk 3]
```diff
@@ -77,16 +81,26 @@ func (b *ResumeJobWorker) Process(message *workers.Msg) {
...
+		b.checkErr(job, err)
 		parsed, err := ParseProcessBatchWorkerMessageArray(pausedJobArr)
-		checkErr(b.Logger, err)
+		b.checkErr(job, err)
 		_, err = b.Workers.CreateProcessBatchJob(parsed.JobID.String(), parsed.AppName, &parsed.Users)
-		checkErr(l, err)
+		b.checkErr(job, err)
 	}
 
+	b.Workers.Statsd.Incr(resumeJobWorkerCompleted, job.Labels(), 1)
```
Comment (bitbanshee) [inline worker/resume_job_worker.go:95]: Should we not cover this return? https://github.com/topfreegames/marathon/pull/51/files#diff-b8c051ba511ba485573808f5be32c13e02b383b281a59b3ca09cd8b501b5a95fR76

**Review by bitbanshee — COMMENTED** [review]
Hey, just remember to have all tests passing :)

File: worker/create_batches_worker.go [hunk 4]
```diff
@@ -252,10 +252,14 @@ func (b *CreateBatchesWorker) Process(message *workers.Msg) {
 		zap.Int("totalParts", msg.TotalParts),
 	)
 
+	b.Workers.Statsd.Incr(CreateBatchesWorkerStart, msg.Job.Labels(), 1)
+
 	err = b.Workers.MarathonDB.Model(&msg.Job).Column("job.status", "App").Where("job.id = ?", msg.Job.ID).Select()
-	checkErr(l, err)
+	b.checkErr(&msg.Job, err)
```
Comment (thiagomachado) [inline worker/create_batches_worker.go:258]: it looks like you've changed the function checkErr to receive a job instead a logger, but maybe you forgot to push the worker/util.go file

Comment (dreis0) [reply worker/create_batches_worker.go:258]: Actually, I've created a `checkErr` function in each worker which will call the `utils` one and also generate an error metric.

**Review by thiagomachado — APPROVED** [review]

**Pull request merged**